pkg/ipcache/metadata.go

```diff
@@ -73,6 +73,12 @@ func GetIDMetadataByIP(prefix string) labels.Labels {
 	return identityMetadata[prefix]
 }
 
+func GetIDMetadata() map[string]labels.Labels {
+	idMDMU.RLock()
+	defer idMDMU.RUnlock()
```

Comment:
The locking here has no effect, as the function returns the map after the lock is released. For the lock to have any effect, you have to copy the map into a new map and return the copy.

Reply:
@twpayne removed the locks since they have no effect. Please resolve these conversations if you think these are fine. Thanks!

Reply:
@chowmean I think you still need the locks as you iterate over `identityMetadata` to create a copy of the map

Something like:

```go
func GetIDMetadata() map[string]labels.Labels {
	idMDMU.RLock()
	defer idMDMU.RUnlock()
        c := make(map[string]labels.Labels, len(identityMetadata))
        for k, v := range identityMetadata {
                  c[k] = v
         }
         return c
```